wrf/utils.py

```diff
@@ -576,26 +591,78 @@ def interp_to_heights(df):
 
     def to_xarray(self,
                   start_time='2013-11-08',time_unit='h',time_step=None,
-                  heights=None,height_var='height',
+                  heights=None,height_var='height',agl=False,
                   exclude=['ts'],
-                  structure='ordered'):
+                  structure='ordered',
+                  **kwargs):
+        """Convert tower time-height data into a xarray dataset.
+        
+        Treatment of the time-varying height coordinates is summarized
+        below:
+
+        heights  height_var  resulting height index
+        -------  ----------  ------------------------------------------
+        None     n/a         levels (Int64Index)
+        None     1d array    assumed constant in time (Float64Index)
+        input    1d array    interpolate to input heights, assuming
+                             constant heights in time (Float64Index)
+        input    2d array    interpolate to input heights at all times
+                             accounting for change in heights over time
+                             (Float64Index)
         
+        Parameters
+        ----------
+        start_time: str or pd.Timestamp
+            The datetime index is constructed from a pd.TimedeltaIndex
+            plus this start_time, where the timedelta index is formed by
+            the saved time array.
+        time_unit: str
+            Timedelta unit for constructing datetime index, only used if
+            time_step is None.
+        time_step: float or None
+            Time-step size, in seconds, to override the output times in
+            the data files. Used in conjunction with start_time to form
+            the datetime index. May be useful if times in output files
+            do not have sufficient precision.
+        heights : array-like or None
+            If None, then use integer levels for the height index,
+            otherwise interpolate to the same heights at all times.
+        height_var : str
+            Name of attribute with actual height values to form the
+            height index. If heights is None, then this must match the
+            number of height levels; otherwise, this may be constant
+            or variable in time.
+        agl : bool, optional
+            Heights by default are specified above sea level; if True,
+            then the "stationz" attribute is used to convert to heights
+            above ground level (AGL).  This only applies if heights are
+            specified.
+        exclude : list
```

Comment:
Out of curiosity, why is this excluded by default? I got it working a little while ago and it contains some variables that are useful for most applications.

Reply:
We can keep all the surface/near-surface quantities in the output. I had originally thought that the data structures were incompatible.

Reply:
@phawbeck Going to leave this as is for now. If you explicitly set `exclude=[]`, it errors out in `to_dataframe()` because there's no "ts" field--all the quantities from the ts file are saved in the tower object as properties. Maybe we could create a separate `to_surface_dataframe()` that would take all of the time-varying surface and near-surface quantities and generate a dataframe from that? That would be simpler, I think, than making it work within the the current `to_dataframe()`, which assumes time-height data.